src/anim/evaluator/anim-target-value.js

```diff
@@ -58,17 +91,48 @@ class AnimTargetValue {
     updateValue(index, value) {
         // always reset the value of the target when the counter is 0
         if (this.counter === 0) {
-            this.value[0] = 0;
-            this.value[1] = 0;
-            this.value[2] = 0;
-            this.value[3] = 1;
+            AnimEvaluator._set(this.value, AnimTargetValue.IDENTITY_QUAT_ARR, this.valueType);
+            if (!this._normalizeWeights) {
+                AnimEvaluator._blend(this.value, this.baseValue, 1, this.valueType);
+            }
         }
-        if (!this.mask[index]) return;
-        if (this.counter === 0) {
-            AnimEvaluator._set(this.value, value, this.valueType);
+        if (!this.mask[index] || this.getWeight(index) === 0) return;
+        if (this._layerBlendType(index) === ANIM_LAYER_ADDITIVE && !this._normalizeWeights) {
+            if (this.valueType === AnimTargetValue.TYPE_QUAT) {
+                // current value
+                const v = AnimTargetValue.q1.set(this.value[0], this.value[1], this.value[2], this.value[3]);
+                // additive value
+                const aV1 = AnimTargetValue.q2.set(this.baseValue[0], this.baseValue[1], this.baseValue[2], this.baseValue[3]);
+                const aV2 = AnimTargetValue.q3.set(value[0], value[1], value[2], value[3]);
+                const aV = aV1.invert().mul(aV2);
+                // scale additive value by it's weight
+                aV.slerp(Quat.IDENTITY, aV, this.getWeight(index));
+                // add the additive value onto the current value then set it to the targets value
+                v.mul(aV);
+                AnimTargetValue.quatArr[0] = v.x;
+                AnimTargetValue.quatArr[1] = v.y;
+                AnimTargetValue.quatArr[2] = v.z;
+                AnimTargetValue.quatArr[3] = v.w;
+                AnimEvaluator._set(this.value, AnimTargetValue.quatArr, this.valueType);
+            } else {
+                AnimTargetValue.v1.set(value[0], value[1], value[2]);
+                AnimTargetValue.v2.set(this.baseValue[0], this.baseValue[1], this.baseValue[2]);
+                const addValue = AnimTargetValue.v1.sub(AnimTargetValue.v2);
```

Comment:
you copy those values to v1 and v2 here just to run sub. I'd probably just do (with having some static addValue Vec3)
`addValue.set(value[0] - this.baseValue[0], value[1] - this.baseValue[1], value[2] - this.baseValue[2]);

Reply:
and then you copy it to `AnimTargetValue.vecArr[0]`
maybe just sub straight to it?
```
AnimTargetValue.vecArr[0] = value[0] - this.baseValue[0];
...
```

Reply:
Sure, I'll update this.